assets/js/notifications.js

```diff
@@ -0,0 +1,173 @@
+const publicVapidKey =
+  "BAK9aUUilxzljIZVaKm8gjt6MWYsXQFbhluMMCFCddHbWug4_H48Q4XtbCwBSPZ9V3wcNUGr92twrEbmGMyABKU";
+const firebaseConfig = {
+  apiKey: "AIzaSyCZ8gbWs9FLlzw93BOj_OQzlWwd_F1N-IY",
+  authDomain: "pyconhyd.firebaseapp.com",
+  databaseURL: "https://pyconhyd.firebaseio.com",
+  projectId: "pyconhyd",
+  storageBucket: "pyconhyd.appspot.com",
+  messagingSenderId: "254977934750",
+  appId: "1:254977934750:web:9f1ad0357425e5718d828f"
+};
+firebase.initializeApp(firebaseConfig);
+const messaging = firebase.messaging();
+messaging.usePublicVapidKey(publicVapidKey);
+
+const tokenDivId = "token_div";
+const permissionDivId = "permission_div";
+
+messaging.onTokenRefresh(() => {
+  messaging
+    .getToken()
+    .then(refreshedToken => {
+      console.log("Token refreshed.");
+      setTokenSentToServer(false);
+      sendTokenToServer(refreshedToken);
+      // Display new Instance ID token and clear UI of all previous messages.
+      resetUI();
+    })
+    .catch(err => {
+      console.log("Unable to retrieve refreshed token ", err);
+      showToken("Unable to retrieve refreshed token ", err);
+    });
+});
+
+messaging.onMessage(payload => {
+  console.log("Message received. ", payload);
+  // Update the UI to include the received message.
+  appendMessage(payload);
+});
+
+function resetUI() {
+  clearMessages();
+  showToken("loading...");
+  messaging
+    .getToken()
+    .then(currentToken => {
+      if (currentToken) {
+        sendTokenToServer(currentToken);
+        updateUIForPushEnabled(currentToken);
+      } else {
+        // Show permission request.
+        console.log(
+          "No Instance ID token available. Request permission to generate one."
+        );
+        // Show permission UI.
+        updateUIForPushPermissionRequired();
+        setTokenSentToServer(false);

```

Comment:
Is this line required ? As per my understanding "sentToServer" should be 1 and it gets set when `resetUI()` is called after getting permission. Why are we setting it to 0 after that using this function call ?

Reply:
We are resetting the variable in storage... once the permission will be received the it is set back again... this is evident in ```updateUIForPushPermissionRequired()``` where ```requestPermission()``` is called

Reply:
Shouldn't this line appear before `updateUIForPushPermissionRequired()` in that case ?